test/framework/prometheus.go

```diff
@@ -315,6 +315,12 @@ func (f *Framework) CreatePrometheusAndWaitUntilReady(ctx context.Context, ns st
 		return nil, fmt.Errorf("waiting for %v Prometheus instances timed out (%v): %v", p.Spec.Replicas, p.Name, err)
 	}
 
+	// We get the prometheus instance again to make sure we have status updated.
```

Comment:
(nit) does it make sense to modify the signature of `WaitForPrometheusReady()` to return the updated object directly?